blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java

```diff
@@ -74,6 +75,52 @@ public String toString() {
         return errors;
     }
 
+    public boolean hasDelayedMetrics(long ingestTime) {
+        if ( getDelayTime(ingestTime) > TRACKER_DELAYED_METRICS_MILLIS ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public long getDelayTime(long ingestTime) {
+        return ingestTime - timestamp;
+    }
+
+    public List<String> getAllMetricNames() {
+        List<String> metricNames = new java.util.ArrayList<String>();
+        if ( gauges != null && gauges.length > 0) {
```

Comment:
I was very tempted to use java 8 stuff. It would've been much cleaner. But that's a decision that I think have external impacts. We need to communicate that better. Can't just cut the cord off, I think...

I even have to change ```String.join()``` to something else, because it's only available on Java 8.